crates/pallet-bridge-pot-currency-swap/src/genesis_verifier.rs

```diff
@@ -0,0 +1,62 @@
+//! An implementation that verifies bridge pot balance values at genesis.
+
+use frame_support::{
+    sp_runtime::{
+        traits::{CheckedAdd, Convert, Zero},
+        ArithmeticError,
+    },
+    sp_std::marker::PhantomData,
+    traits::{Currency, Get},
+};
+
+use super::{Config, GenesisVerifier, Pallet};
+
+/// The implementation that requires bridge pot balance values be balanced.
+pub struct Balanced<Pallet>(PhantomData<Pallet>);
+
+/// The bridge pot balances logic related error kinds.
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// The arithmetic operation error.
+    Arithmetic(ArithmeticError),
+}
+
+impl<T: Config<I>, I: 'static> Balanced<Pallet<T, I>> {
+    /// A function to calculate expected [`Config::PotTo`] balance based on the provided list of
+    /// all [`Config::AccountIdFrom`] balances except [`Config::PotFrom`] balance.
+    pub fn calculate_expected_to_bridge_balance(
```

Comment:
```suggestion
    pub fn calculate_expected_bridge_balance(
```

Although specific, the `to` bit here is confusing. I propose we drop it, as it is clear from the types which bridge the balance is for.